File: javascript/7_kyu/simple_string_characters.js
// Simple string characters
// https://www.codewars.com/kata/5a29a0898f27f2d9c9000058/train/javascript

function solve(s){
    var u = 0;
    var l = 0;
    var n = 0;
    var special = 0;
    var letters = /^[A-Za-z]+$/;
    
    for (var i = 0; i < s.length; i++) {
      if (s[i].match(letters) && s[i] == s[i].toUpperCase()) {
          u++;
      } else if (s[i].match(letters) && s[i] == s[i].toLowerCase()) {
          l++;
      } else if ( !(isNaN(s[i])) ) {
          n++;
      } else {
          special++;
      }
    }

    return [u, l, n, special];
  }

console.log(solve("Codewars@codewars123.com"));

File: python/7_kyu/simple_string_characters.py
# Simple string characters
# https://www.codewars.com/kata/5a29a0898f27f2d9c9000058/train/python

def solve(s):
    upper = 0
    lower = 0
    number = 0
    special = 0
    
    for i in s:
        if i.isalpha() and i.isupper():
            upper += 1
        elif i.isalpha() and i.islower():
            lower += 1
        elif i.isnumeric():
            number += 1
        else:
            special += 1
    
    return [upper, lower, number, special]